src/pcre2_compile.c

```diff
@@ -1940,19 +1943,19 @@ else
       ptr--;   /* Back to the digit */
 
       /* As we know we are at a digit, the only possible error from
-      read_number() is a number that is too large to be a group number. In this
-      case we treat the group number as too-large (since it may be larger than
-      INT_MAX we cannot return it for the caller to check).
+      read_number() is a number that is too large to be a group number. Because
+      that number might be still valid if read as an octal, errorcodeptr is not
+      set on failure, and therefore we rely on its value being slightly
+      bigger than MAX_GROUP_NUMBER to set it, if not falling through. */
 
-      \1 to \9 are always back references. \8x and \9x are too; \1x to \7x
-      are octal escapes if there are not that many previous captures. */
+      read_number(&ptr, ptrend, -1, MAX_GROUP_NUMBER, 0, &s, errorcodeptr);
```

Comment:
You changed the code since I reviewed. You're now ignoring the return value from `read_number`, and baking in the assumption that when `read_number` returns false, `s` will be populated with the partially-parsed value.

That seems nasty. You should keep the check on its return value.

Reply:
> `s` will be populated with the partially-parsed value.

the partially parsed value, but one that would be slightly larger than  MAX_GROUP_NUMBER which will be returned by the function in those cases.

> That seems nasty. You should keep the check on its return value

OK, but then I would also document better how the IMHO equally bogus value we are going to use is relevant to the code below.

Reply:
I'd prefer to think of it as a "sentinel value" than a "bogus value". 😉 

But thank you for updating the comments.

I'm happy knowing that the code can't be broken accidentally by refactoring, especially with the assertion you added.